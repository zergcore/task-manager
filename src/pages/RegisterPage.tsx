import { Container, Row, Col, Card } from "react-bootstrap";
import { Link } from "react-router-dom";
import RegisterForm from "../components/auth/RegisterForm";

const RegisterPage = () => {
  return (
    <Container className="py-5">
      <Row>
        <Col md={6} className="mx-auto">
          <Card>
            <Card.Body className="p-4">
              <RegisterForm />
              <div className="text-center mt-3">
                <p>
                  Already have an account? <Link to="/login">Login here</Link>
                </p>
              </div>
            </Card.Body>
          </Card>
        </Col>
      </Row>
    </Container>
  );
};

export default RegisterPage;
