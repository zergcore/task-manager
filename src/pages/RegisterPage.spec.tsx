import { render } from '@testing-library/react';
import RegisterPage from './RegisterPage';
import { MemoryRouter } from 'react-router-dom';
import { describe, test, expect } from 'vitest'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

describe('RegisterPage', () => {
  test('should match snapshot', () => {
    const queryClient = new QueryClient();
    const { asFragment } = render(
      <QueryClientProvider client={queryClient}>
        <MemoryRouter>
          <RegisterPage />
        </MemoryRouter>
      </QueryClientProvider>

    );

    expect(asFragment()).toMatchSnapshot();
  });
});