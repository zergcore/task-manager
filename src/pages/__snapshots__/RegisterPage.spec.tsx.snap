// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RegisterPage > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="py-5 container"
  >
    <div
      class="row"
    >
      <div
        class="mx-auto col-md-6"
      >
        <div
          class="card"
        >
          <div
            class="p-4 card-body"
          >
            <form
              class=""
            >
              <h2
                class="mb-4"
              >
                Create a Task Manager Account
              </h2>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                >
                  Email
                </label>
                <input
                  class="form-control"
                  name="email"
                  required=""
                  type="email"
                  value=""
                />
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  minlength="6"
                  name="password"
                  required=""
                  type="password"
                  value=""
                />
                <small
                  class="text-muted form-text"
                >
                  Password must be at least 6 characters long
                </small>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary w-100 css-74d805-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Register
              </button>
            </form>
            <div
              class="text-center mt-3"
            >
              <p>
                Already have an account? 
                <a
                  data-discover="true"
                  href="/login"
                >
                  Login here
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
